File: go/06-io-operations/formatted_output.go
package main

import "fmt"

func main() {
	name := "Ada"
	age := 37
	pi := 3.14159

	// Println adds spaces between operands and a trailing newline
	fmt.Println("Hello,", name)

	// Print adds no newline and no spaces between string operands
	fmt.Print("Age: ")
	fmt.Print(age)
	fmt.Print("\n")

	// Printf uses format verbs for precise control
	fmt.Printf("%s is %d years old\n", name, age)
	fmt.Printf("Pi rounded: %.2f\n", pi)
	fmt.Printf("Hex: %x, Binary: %b\n", 255, 5)

	// Sprintf returns a formatted string instead of printing it
	greeting := fmt.Sprintf("Welcome, %s!", name)
	fmt.Println(greeting)
}

File: r/06-io-operations/formatted_output.R
# Formatted output in R

# print() shows the [1] index prefix and quotes strings
print("Statistical computing")
print(c(1.5, 2.5, 3.5))

# cat() joins values with a space separator, no quotes or index
# fill = TRUE adds a trailing newline for us
cat("Values:", 1, 2, 3, fill = TRUE)

# sprintf() gives C-style format control
cat(sprintf("Integer: %d\n", 42L))
cat(sprintf("Float:   %.3f\n", 2 / 3))
cat(sprintf("Percent: %5.1f%%\n", 87.5))
cat(sprintf("Zero-pad: %08.2f\n", 3.14))

# paste() builds strings; paste0() uses no separator
label <- paste("R", "version", "4.4.2")
cat(label, fill = TRUE)
cat(paste0("file", "_", "name", ".csv"), fill = TRUE)
